### PR TITLE
Keep abundaces in envelope at collapse & M4,mu4 values

### PR DESCRIPTION
**Pull request by ezapartas**

### CODE REVIEW

File: posydon/binary_evol/SN/profile_collapse.py [hunk 1]
```diff
@@ -24,6 +24,40 @@
     'Aldo Batta <aldobatta@gmail.com>',
 ]
 
+def get_ejecta_element_mass_at_collapse(star, compact_object_mass, verbose):
+    # read star quantities
+    enclosed_mass_all = star.profile['mass'][::-1]*Mo  # cell outer total mass
```
Comment (mkruckow) [inline posydon/binary_evol/SN/profile_collapse.py]: `Mo` should be replaced by `const.Msun`.

File: posydon/binary_evol/SN/profile_collapse.py [hunk 2]
```diff
@@ -24,6 +24,40 @@
...
+    # shell's mass
+    dm_all = enclosed_mass[1:] - enclosed_mass[:-1]
+
+    if 'h1' in star.profile.dtype.names:
+        h1_all = star.profile['h1'][::-1]  # h1 mass fraction
+    if 'he4' in star.profile.dtype.names:
+        he4_all = star.profile['he4'][::-1]  # he4 mass fraction
+    if 'o16' in star.profile.dtype.names:
+        o16_all = star.profile['o16'][::-1]  # he4 mass fraction
```
Comment (mkruckow) [inline posydon/binary_evol/SN/profile_collapse.py]: Here is the comment not updated to `o16`.

File: posydon/binary_evol/SN/profile_collapse.py [hunk 3]
```diff
@@ -24,6 +24,40 @@
     'Aldo Batta <aldobatta@gmail.com>',
 ]
 
+def get_ejecta_element_mass_at_collapse(star, compact_object_mass, verbose):
```
Comment (mkruckow) [inline posydon/binary_evol/SN/profile_collapse.py:27]: We should add a docstring to the new function, e.g.
```
    """Calculate the masses of H1, He4, and O16 in the ejecta.
    
    Parameters
    ----------
    star : object
        Star object of a collapsing star containing the MESA profile.
    compact_object_mass : float
        The mass of the compact object (in Msun), hence all above this mass will be ejected.
    verbose : bool
        If `True`, it prints some informations.
    
    Returns
    -------
    h1_mass_ej : float
        Hydrogen mass in the ejecta. (in Msun)
    he4_mass_ej : float
        Helium mass in the ejecta. (in Msun)
    o16_mass_ej : float
        Oxygen mass in the ejecta. (in Msun)
    """
```

File: posydon/binary_evol/SN/profile_collapse.py (hunk 1 above)
Comment (mkruckow) [reply posydon/binary_evol/SN/profile_collapse.py]: I think, we need to go once more carefully through the unit changes to make sure all is as it should.

**Comment by mkruckow** [comment]
There are some more files to check, whether to add `h1_mass_ej`, `he4_mass_ej`, and `o16_mass_ej` there:

- `posydon/binary_evol/MESA/step_mesa.py`: putting the interpolated values to the star
- `posydon/binary_evol/singlestar.py`: putting the interpolated values to the star
- `posydon/interpolation/interpolation.py`: add to core collapse keys
- `posydon/popsyn/io.py`: set a default column format for pandas tables written to hdf5 files
- `posydon/visualization/plot_defaults.py`: set a default for plotting the new values

File: posydon/binary_evol/SN/profile_collapse.py (hunk 1 above)
Comment (ezapartas) [reply posydon/binary_evol/SN/profile_collapse.py]: I think that grams are not needed at all in the function in the end, so everything should be in Msun now

**Comment by astroJeff** [comment]
@ezapartas, this is an interesting addition to our code. Let's discuss next Thursday. I was concerned with the same thing that @mkruckow pointed out about whether there are other files that also need to be adjusted, but he did the work to find exactly which files also need to be adjusted.

**Review by mkruckow — DISMISSED** [review]
Looks good to me.

**Review by mkruckow — DISMISSED** [review]

**Review by maxbriel — DISMISSED** [review]
Similar comment about #309 compatibility and again requires retraining of the interpolators.

**Review by mkruckow — APPROVED** [review]
I have made another test on the CO-HMS grids on yggdrasil and all looked fine to me.

**Review by astroJeff — APPROVED** [review]

**Pull request merged**